app/population/middleware/set-results.population-table.js

```diff
@@ -0,0 +1,30 @@
+const presenters = require('../../../common/presenters')
+const locationsFreeSpacesService = require('../../../common/services/locations-free-spaces')
+
+async function setResultsPopulationTable(req, res, next) {
+  try {
+    const { dateRange, filter } = req
+
+    const freeSpaces = await locationsFreeSpacesService.getPrisonFreeSpaces({
+      dateFrom: dateRange[0],
+      dateTo: dateRange[1],
+      filter,
```

Comment:
So, I think this usage is slightly different from out other cases.

The `req.filter` value is actually used to build the filter component that ultimately gets passed down to the view. And then the filter and results middlewares share a unified `req.body` value that they use to call their respective services.

I'd say in this case, if you don't need to build the filter component you can do away with `set-body.{thing}.js` and `set-filter.{thing}.js` and just use the values directly in the `set-results.{thing}.js` middlware.

Here is [an example of a set-body middleware](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/master/app/allocations/middleware/set-body.allocations.js). You'll notice that's where the locations and dates are pulled from the the querystring or params and used to build the body call sent to the services. And then both respective [set-filter middleware](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/ef40ccf4dbe16dc4919e204ea43d9c8c250e8a3f/app/allocations/middleware/set-filter.allocations.js#L12) and [set-results middleware](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/master/app/allocations/middleware/set-results.allocations.js#L14) use this shared body value.

So (if you don't need a filter component) I think you should just do the following in set-results middleware and remove the need for the set-filter middleware:

```javascript
const presenters = require('../../../common/presenters')
const locationsFreeSpacesService = require('../../../common/services/locations-free-spaces')

async function setResultsPopulationTable(req, res, next) {
  try {
    const { locations, dateRange, query } = req
    const freeSpaces = await locationsFreeSpacesService.getPrisonFreeSpaces({
      dateFrom: dateRange[0],
      dateTo: dateRange[1],
      locationIds: locations?.join(),
    })

    req.resultsAsPopulationTable = presenters.locationsToPopulationTableComponent(
      {
        query,
        startDate: dateRange[0],
        focusDate: new Date(),
      }
    )(freeSpaces)

    next()
  } catch (error) {
    next(error)
  }
}

module.exports = setResultsPopulationTable
```

Reply:
Ok, so that was a misunderstanding of how all these bits work, I'll get this information added into the module PR (#870).

In the meantime I've refactored the population filter middleware out of existence, and made the suggestion as above